### PR TITLE
[DEVHAS-68] Don't modify Component CR metadata

### PR DESCRIPTION
**Pull request by johnmcollier**
Updates HAS so that the Component controller does not attach labels and annotations to the resource during Component creation, which avoids an unnecessary reconciliation.

- Adds information about the Component's GitOps repository to the Component
   - Repository URL
   - Branch (if specified)
   - Context (if specified)
- Removes the code that sets the labels and annotations on the Component CR
- Updates the tests accordingly

### CODE REVIEW

File: controllers/component_controller.go [hunk 1]
```diff
@@ -535,20 +522,17 @@ func (r *ComponentReconciler) generateGitops(component *appstudiov1alpha1.Compon
 	return r.AppFS.RemoveAll(tempDir)
 }
 
-// setGitopsAnnotations adds the necessary gitops annotations (url, branch, context) to the component CR object
-func setGitopsAnnotations(component *appstudiov1alpha1.Component, devfileData data.DevfileData) error {
+// setGitopsStatus adds the necessary gitops annotations (url, branch, context) to the component CR status
```
Comment (maysunfaisal) [inline controllers/component_controller.go:525]:
```suggestion
// setGitopsStatus adds the necessary gitops info (url, branch, context) to the component CR status
```

File: controllers/component_controller_test.go [hunk 2]
```diff
@@ -1034,7 +914,7 @@ var _ = Describe("Component controller", func() {
 			createdHasComp := &appstudiov1alpha1.Component{}
 			Eventually(func() bool {
 				k8sClient.Get(context.Background(), hasCompLookupKey, createdHasComp)
-				return len(createdHasComp.Status.Conditions) > 0 && createdHasComp.GetLabels()["appstudio.has/component"] != ""
+				return len(createdHasComp.Status.Conditions) > 0 //&& createdHasComp.GetLabels()["appstudio.has/component"] != ""
```
Comment (maysunfaisal) [inline controllers/component_controller_test.go:917]: should we remove comment?

File: controllers/component_controller_test.go [hunk 3]
```diff
@@ -1186,19 +1071,21 @@ var _ = Describe("Component controller", func() {
...
```
Comment (maysunfaisal) [inline controllers/component_controller_test.go:1074]: here too

**Review by maysunfaisal — APPROVED** [review]
some nits about comments, can fix them later too, dont mind this being merged

File: controllers/component_controller_test.go (hunk 2 above)
Comment (johnmcollier) [reply controllers/component_controller_test.go:917]: Hard to explain - will address in a follow up PR

**Pull request merged**